src/winkerberos.c

```diff
@@ -359,6 +364,11 @@ sspi_client_init(PyObject* self, PyObject* args, PyObject* kw) {
         ulen = wcslen(user);
     }
 
+    if (mechoidobj != NULL && PyCapsule_CheckExact(mechoidobj)) {
```

Comment:
We have to use PyCObject functions (or the aliases defined in this file). This should be something like:

```
if (mechoidobj != NULL) {
    if (!PyCObject_Check(mechoidobj)) {
        PyErr_SetString(PyExc_TypeError, "Expected a context object for mech_oid");
        goto done;
    } else {
        mechoid = (WCHAR*)PyCObject_AsVoidPtr(mechoidobj);
        if (mechoid == NULL) {
            goto done;
        }
    }
}
```